### DISCUSSION TITLE
Next steps?

### DISCUSSION THREAD

**Discussion by erikbra** [grate-devs/grate#111] [Q&A]
I tried to look through the RoundhousE issues, and looked at the grate codebase, and I couldn't find much of RoundhousE that isn't supported now in grate. What do others think? Are there any actual features missing from grate that RoundhousE did, that we haven't written off at "not gonna fix", or some issues from the RoundhousE repo that we should consider implementing in grate?

In my head, documentation is what could be improved most now. I'll try to go through, and read more RoundhousE docs, and see if there is any more areas we should cover. Maybe even more "what is grate" stuff? I'm not sure.

Are there any other ideas/opinions?

**Comment by bptillman** [comment]
I've been quiet as of late after you invited me to this, but I'd agree that documentation is the key thing. With the goal being a drop in replacement for Roundhouse, I think it would be a good idea to make sure anything relevant that is mentioned in their docs is also mentioned here. I did a quick look over their docs, and here is a few things I found in their documentation that is not here (yet):

- What databases are supported
- Any prerequisites
- On Directory Run Order, including sample output that would be seen at each step
- [Configuration Options](https://github.com/chucknorris/roundhouse/wiki/ConfigurationOptions)
- Mention limitations of Grate - This could be similar to the page that already exists talking about how it different from Roundhouse, but may be having one more generalized to things Grate does not do that other db migration tools do. [Here is the Roundhouse page for reference](https://github.com/chucknorris/roundhouse/wiki/Features#limitations-of-rh) 

Additionally, there are a few open issues around documentation that I think would be good to get done. Namely #68 , #116 , and #117  .

Outside of documentation I haven't seen any features that Roundhouse has that you've said wont be in Grate that are show stoppers for me. But confirming things work, like #113  and #115 would be a good idea. The only one that does not exist in Grate yet that I have seen used (and I might try to take a stab at implementing) is the ability to restore from a db backup and then migrate it.

What do you think about those ideas?

**Reply by wokket** [reply]
I'm also keen to see the first class docco, hence trying to get it progressed...  It's really hard to convince teams to move to a third-party, open source tool (even if it's superior) without quality docco, otherwise they (understandably) tend to stick to things like EF migrations purely from a "what happens when you're gone and it doesn't work" perspective.
